### PR TITLE
docs: move and rewrite bake reference

### PR DESCRIPTION
**Pull request by dvdksn**
Signed-off-by: David Karlsson <david.karlsson@docker.com>

- Move the bake file reference to `docker/buildx` repo to improve maintainability
- Rewrite the doc to add more examples

### CODE REVIEW

**Comment by dvdksn** [comment]
@crazy-max I think I managed to migrate the file with history... But I suppose the DCO issue can't really be avoided since many of those commits weren't signed.

**Comment by dvdksn** [comment]
> [Error: Unexpected HTTP response: 429](https://github.com/docker/buildx/actions/runs/4758021022/jobs/8455545085?pr=1748#step:3:105)

**Comment by crazy-max** [comment]
> @crazy-max I think I managed to migrate the file with history... But I suppose the DCO issue can't really be avoided since many of those commits weren't signed.

That's fine for DCO, this is expected.

> > [Error: Unexpected HTTP response: 429](https://github.com/docker/buildx/actions/runs/4758021022/jobs/8455545085?pr=1748#step:3:105)

Looks like a network issue, let me restart the build.

**Review by crazy-max — APPROVED** [review]

File: docs/bake-reference.md [hunk 1]
```diff
@@ -0,0 +1,743 @@
...
+      "context": ".",
+      "dockerfile": "Dockerfile"
+    }
+  }
+}
+```
+
+### `target.contexts`
+
+Additional build contexts.
```
Comment (tonistiigi) [inline docs/bake-reference.md:250]: Follow up: Seems like some things are missing. https://github.com/docker/buildx/pull/928/commits/c3dab802d8eacc546a7c9328cdfcd7ee363f870a

**Review by tonistiigi — APPROVED** [review]

**Comment by dvdksn** [comment]
Added a missing section built-in variables

File: docs/bake-reference.md
Comment (jedevc) [inline docs/bake-reference.md:1]: Could we also pull in https://github.com/docker/docs/pull/17041?

Maybe as a follow-up, could we also maybe link to the list of builtin functions? https://github.com/docker/buildx/blob/master/bake/hclparser/stdlib.go#L17-L116. Don't know how best to manage that, ideally we wouldn't need to write and maintain docs for *every* single function.

Comment (dvdksn) [reply docs/bake-reference.md:1]: There's already a [link to `stdlib.go`](https://github.com/docker/buildx/pull/1748/files#diff-f61018ac93cb764173f498ca6d0f2c27321c103e43aa7d424ee61a55a101232bR760)

Comment (jedevc) [reply docs/bake-reference.md:1]: :facepalm: ugh sorry. My bad :pray:

Comment (jedevc) [reply docs/bake-reference.md:1]: Thaaaanks :heart:

**Comment by crazy-max** [comment]
docs-upstream job does not look happy: https://github.com/docker/buildx/actions/runs/4808382935/jobs/8558351051?pr=1748#step:9:4001

```
#29 [htmltest htmltest 2/2] RUN <<EOF (cat /results...)
#0 0.057 htmltest started at 12:04:18 on _site
#0 0.057 ========================================================================
#0 0.057 engine/reference/commandline/buildx_bake/index.html
#0 0.057   target does not exist --- engine/reference/commandline/buildx_bake/index.html --> /build/bake/reference/
#0 0.057 ========================================================================
#0 0.057 ✘✘✘ failed in 2.800544925s
#0 0.057 1 errors in 2312 documents
```

I guess we need to fix upstream first?

**Review by crazy-max — APPROVED** [review]

File: docs/bake-reference.md
Comment (jedevc) [reply docs/bake-reference.md:1]: Split these out into https://github.com/docker/buildx/pull/1791

**Pull request merged**